File: src/extras-kwidgetsaddons/libkratingpainter.cpp
#include <KRatingPainter>
#include <QIcon>
#include <QPainter>
#include <QPixmap>
#include <QPoint>
#include <QRect>
#include <kratingpainter.h>
#include "libkratingpainter.h"
#include "libkratingpainter.hxx"

KRatingPainter* KRatingPainter_new() {
    return new KRatingPainter();
}

int KRatingPainter_MaxRating(const KRatingPainter* self) {
    return self->maxRating();
}

bool KRatingPainter_HalfStepsEnabled(const KRatingPainter* self) {
    return self->halfStepsEnabled();
}

int KRatingPainter_Alignment(const KRatingPainter* self) {
    return static_cast<int>(self->alignment());
}

int KRatingPainter_LayoutDirection(const KRatingPainter* self) {
    return static_cast<int>(self->layoutDirection());
}

QIcon* KRatingPainter_Icon(const KRatingPainter* self) {
    return new QIcon(self->icon());
}

bool KRatingPainter_IsEnabled(const KRatingPainter* self) {
    return self->isEnabled();
}

QPixmap* KRatingPainter_CustomPixmap(const KRatingPainter* self) {
    return new QPixmap(self->customPixmap());
}

int KRatingPainter_Spacing(const KRatingPainter* self) {
    return self->spacing();
}

void KRatingPainter_SetMaxRating(KRatingPainter* self, int max) {
    self->setMaxRating(static_cast<int>(max));
}

void KRatingPainter_SetHalfStepsEnabled(KRatingPainter* self, bool enabled) {
    self->setHalfStepsEnabled(enabled);
}

void KRatingPainter_SetAlignment(KRatingPainter* self, int alignVal) {
    self->setAlignment(static_cast<Qt::Alignment>(alignVal));
}

void KRatingPainter_SetLayoutDirection(KRatingPainter* self, int direction) {
    self->setLayoutDirection(static_cast<Qt::LayoutDirection>(direction));
}

void KRatingPainter_SetIcon(KRatingPainter* self, const QIcon* icon) {
    self->setIcon(*icon);
}

void KRatingPainter_SetEnabled(KRatingPainter* self, bool enabled) {
    self->setEnabled(enabled);
}

void KRatingPainter_SetCustomPixmap(KRatingPainter* self, const QPixmap* pixmap) {
    self->setCustomPixmap(*pixmap);
}

void KRatingPainter_SetSpacing(KRatingPainter* self, int spacing) {
    self->setSpacing(static_cast<int>(spacing));
}

void KRatingPainter_Paint(const KRatingPainter* self, QPainter* painter, const QRect* rect, int rating) {
    self->paint(painter, *rect, static_cast<int>(rating));
}

int KRatingPainter_RatingFromPosition(const KRatingPainter* self, const QRect* rect, const QPoint* pos) {
    return self->ratingFromPosition(*rect, *pos);
}

void KRatingPainter_PaintRating(QPainter* p, const QRect* rect, int alignVal, int rating) {
    KRatingPainter::paintRating(p, *rect, static_cast<Qt::Alignment>(alignVal), static_cast<int>(rating));
}

int KRatingPainter_GetRatingFromPosition(const QRect* rect, int alignVal, int direction, const QPoint* pos) {
    return KRatingPainter::getRatingFromPosition(*rect, static_cast<Qt::Alignment>(alignVal), static_cast<Qt::LayoutDirection>(direction), *pos);
}

void KRatingPainter_Paint4(const KRatingPainter* self, QPainter* painter, const QRect* rect, int rating, int hoverRating) {
    self->paint(painter, *rect, static_cast<int>(rating), static_cast<int>(hoverRating));
}

void KRatingPainter_PaintRating5(QPainter* p, const QRect* rect, int alignVal, int rating, int hoverRating) {
    KRatingPainter::paintRating(p, *rect, static_cast<Qt::Alignment>(alignVal), static_cast<int>(rating), static_cast<int>(hoverRating));
}

void KRatingPainter_Delete(KRatingPainter* self) {
    delete self;
}

File: src/extras-kwidgetsaddons/libkratingpainter.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qnamespace_enums = @import("../libqnamespace.zig").enums;

/// https://api-staging.kde.org/kratingpainter.html
pub const kratingpainter = struct {
    /// New constructs a new KRatingPainter object.
    ///
    ///
    pub fn New() QtC.KRatingPainter {
        return qtc.KRatingPainter_new();
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#maxRating)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn MaxRating(self: ?*anyopaque) i32 {
        return qtc.KRatingPainter_MaxRating(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#halfStepsEnabled)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn HalfStepsEnabled(self: ?*anyopaque) bool {
        return qtc.KRatingPainter_HalfStepsEnabled(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#alignment)
    ///
    /// ``` self: QtC.KRatingPainter ```
    ///
    /// Returns: ``` flag of qnamespace_enums.AlignmentFlag ```
    pub fn Alignment(self: ?*anyopaque) i32 {
        return qtc.KRatingPainter_Alignment(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#layoutDirection)
    ///
    /// ``` self: QtC.KRatingPainter ```
    ///
    /// Returns: ``` qnamespace_enums.LayoutDirection ```
    pub fn LayoutDirection(self: ?*anyopaque) i32 {
        return qtc.KRatingPainter_LayoutDirection(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#icon)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn Icon(self: ?*anyopaque) QtC.QIcon {
        return qtc.KRatingPainter_Icon(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#isEnabled)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn IsEnabled(self: ?*anyopaque) bool {
        return qtc.KRatingPainter_IsEnabled(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#customPixmap)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn CustomPixmap(self: ?*anyopaque) QtC.QPixmap {
        return qtc.KRatingPainter_CustomPixmap(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#spacing)
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn Spacing(self: ?*anyopaque) i32 {
        return qtc.KRatingPainter_Spacing(@ptrCast(self));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setMaxRating)
    ///
    /// ``` self: QtC.KRatingPainter, max: i32 ```
    pub fn SetMaxRating(self: ?*anyopaque, max: i32) void {
        qtc.KRatingPainter_SetMaxRating(@ptrCast(self), @intCast(max));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setHalfStepsEnabled)
    ///
    /// ``` self: QtC.KRatingPainter, enabled: bool ```
    pub fn SetHalfStepsEnabled(self: ?*anyopaque, enabled: bool) void {
        qtc.KRatingPainter_SetHalfStepsEnabled(@ptrCast(self), enabled);
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setAlignment)
    ///
    /// ``` self: QtC.KRatingPainter, alignVal: flag of qnamespace_enums.AlignmentFlag ```
    pub fn SetAlignment(self: ?*anyopaque, alignVal: i32) void {
        qtc.KRatingPainter_SetAlignment(@ptrCast(self), @intCast(alignVal));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setLayoutDirection)
    ///
    /// ``` self: QtC.KRatingPainter, direction: qnamespace_enums.LayoutDirection ```
    pub fn SetLayoutDirection(self: ?*anyopaque, direction: i32) void {
        qtc.KRatingPainter_SetLayoutDirection(@ptrCast(self), @intCast(direction));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setIcon)
    ///
    /// ``` self: QtC.KRatingPainter, icon: QtC.QIcon ```
    pub fn SetIcon(self: ?*anyopaque, icon: ?*anyopaque) void {
        qtc.KRatingPainter_SetIcon(@ptrCast(self), @ptrCast(icon));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setEnabled)
    ///
    /// ``` self: QtC.KRatingPainter, enabled: bool ```
    pub fn SetEnabled(self: ?*anyopaque, enabled: bool) void {
        qtc.KRatingPainter_SetEnabled(@ptrCast(self), enabled);
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setCustomPixmap)
    ///
    /// ``` self: QtC.KRatingPainter, pixmap: QtC.QPixmap ```
    pub fn SetCustomPixmap(self: ?*anyopaque, pixmap: ?*anyopaque) void {
        qtc.KRatingPainter_SetCustomPixmap(@ptrCast(self), @ptrCast(pixmap));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#setSpacing)
    ///
    /// ``` self: QtC.KRatingPainter, spacing: i32 ```
    pub fn SetSpacing(self: ?*anyopaque, spacing: i32) void {
        qtc.KRatingPainter_SetSpacing(@ptrCast(self), @intCast(spacing));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#paint)
    ///
    /// ``` self: QtC.KRatingPainter, painter: QtC.QPainter, rect: QtC.QRect, rating: i32 ```
    pub fn Paint(self: ?*anyopaque, painter: ?*anyopaque, rect: ?*anyopaque, rating: i32) void {
        qtc.KRatingPainter_Paint(@ptrCast(self), @ptrCast(painter), @ptrCast(rect), @intCast(rating));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#ratingFromPosition)
    ///
    /// ``` self: QtC.KRatingPainter, rect: QtC.QRect, pos: QtC.QPoint ```
    pub fn RatingFromPosition(self: ?*anyopaque, rect: ?*anyopaque, pos: ?*anyopaque) i32 {
        return qtc.KRatingPainter_RatingFromPosition(@ptrCast(self), @ptrCast(rect), @ptrCast(pos));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#paintRating)
    ///
    /// ``` p: QtC.QPainter, rect: QtC.QRect, alignVal: flag of qnamespace_enums.AlignmentFlag, rating: i32 ```
    pub fn PaintRating(p: ?*anyopaque, rect: ?*anyopaque, alignVal: i32, rating: i32) void {
        qtc.KRatingPainter_PaintRating(@ptrCast(p), @ptrCast(rect), @intCast(alignVal), @intCast(rating));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#getRatingFromPosition)
    ///
    /// ``` rect: QtC.QRect, alignVal: flag of qnamespace_enums.AlignmentFlag, direction: qnamespace_enums.LayoutDirection, pos: QtC.QPoint ```
    pub fn GetRatingFromPosition(rect: ?*anyopaque, alignVal: i32, direction: i32, pos: ?*anyopaque) i32 {
        return qtc.KRatingPainter_GetRatingFromPosition(@ptrCast(rect), @intCast(alignVal), @intCast(direction), @ptrCast(pos));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#paint)
    ///
    /// ``` self: QtC.KRatingPainter, painter: QtC.QPainter, rect: QtC.QRect, rating: i32, hoverRating: i32 ```
    pub fn Paint4(self: ?*anyopaque, painter: ?*anyopaque, rect: ?*anyopaque, rating: i32, hoverRating: i32) void {
        qtc.KRatingPainter_Paint4(@ptrCast(self), @ptrCast(painter), @ptrCast(rect), @intCast(rating), @intCast(hoverRating));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#paintRating)
    ///
    /// ``` p: QtC.QPainter, rect: QtC.QRect, alignVal: flag of qnamespace_enums.AlignmentFlag, rating: i32, hoverRating: i32 ```
    pub fn PaintRating5(p: ?*anyopaque, rect: ?*anyopaque, alignVal: i32, rating: i32, hoverRating: i32) void {
        qtc.KRatingPainter_PaintRating5(@ptrCast(p), @ptrCast(rect), @intCast(alignVal), @intCast(rating), @intCast(hoverRating));
    }

    /// [Qt documentation](https://api-staging.kde.org/kratingpainter.html#dtor.KRatingPainter)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KRatingPainter ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KRatingPainter_Delete(@ptrCast(self));
    }
};
